src/linking.jl

```diff
@@ -0,0 +1,75 @@
+function add_linking_constraints(m::Optimizer, λ::Dict)
+    """
+    Add linking constraints between λ[i], λ[j] variables.
+
+    Reference information:
+        Jongeun Kim, Jean-Philippe P. Richard, Mohit Tawarmalani, Piecewise Polyhedral Relaxations of Multilinear Optimization, 
+        http://www.optimization-online.org/DB_HTML/2022/07/8974.html
+
+    For example, suppose we have λ[i], λ[j], and λ[k] where i=(1,2,3), j=(1,2,4), and k=(1,2,5).
+    λ[i] contains all multipliers for the extreme points in the space of (x1,x2,x3). 
+    λ[j] contains all multipliers for the extreme points in the space of (x1,x2,x4).
+    λ[k] contains all multipliers for the extreme points in the space of (x1,x2,x5).
+
+    Using λ[i], λ[j], or λ[k], we can express multilinear function x1*x2.
+    We define a linking variable μ(1,2) that represents the value of x1*x2.
+    Linking constraints are
+        μ(1,2) == convex combination expression for x1*x2 using λ[i],    
+        μ(1,2) == convex combination expression for x1*x2 using λ[j], and   
+        μ(1,2) == convex combination expression for x1*x2 using λ[k].    
+
+    Then, these constraints link between λ[i], λ[j], and λ[k] variables.
+    """
+
+    # (Step 0) Define informational paramters used in the report in the end of the function.
+    t_begin = time()
+    num_added_constraints = 0
+
+    # (Step 1) Define link_info::Dict 
+    #   - key: linking variable indices
+    #   - value: the subset of key(λ) where each element is a superset of the linking variable indices 
+    maxdeg = maximum([length(k) for k in keys(λ)]) # Compute maximum degree of multilinear terms
+    if maxdeg <= 2
+        # @info "no linking constraint is added because the maximum degree of multilinear term is $(maxdeg)."
+        return
+    end
+    all_vars_idx = sort(collect(union(collect(keys(λ))...))) # Collect all variable indices used in multilinear terms
+    link_info = Dict(
+        link_var_idx => filter(r -> issubset(link_var_idx, r), keys(λ)) for
+        deg in 2:maxdeg-1 for
+        link_var_idx in Combinatorics.combinations(all_vars_idx, deg)
+    )
+    filter!(r -> length(r.second) >= 2, link_info)
```

Comment:
Can potentially save `link_info` inside the model so that this computation (expensive) can be avoided at every iteration.